### PR TITLE
DO-59 Provision Redis 5.x for persisting states for the Coordinator

### PR DESCRIPTION
**Pull request by rsaffi**
### References

[PB-353](https://xainag.atlassian.net/browse/PB-353) -> [DO-59](https://xainag.atlassian.net/browse/DO-59)

### Summary

Updates `docker-compose-dev.yml` and `docker-compose.yml` for provisioning Redis 5.x.
Redis will be used as in-memory key-value storage for the Coordinator to persist its state externally.

### Are there any open tasks/blockers for the ticket?

No

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [ ] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [ ] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [ ] Linked the ticket in the merge request title or the references section.
- [ ] Added an informative merge request summary.

**Code checklist**

- [ ] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [ ] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -19,6 +19,17 @@ services:
...
       - "9000"
 
+  redis:
+    image: redis:5 # current stable version, v6 out between March 2020 and May 2020
+    hostname: redis
+    entrypoint: /usr/local/bin/redis-server --appendonly yes --appendfsync everysec # using combination of RDB and AOF for persistence: https://redis.io/topics/persistence
+    volumes:
+      - redis-data:/data
+    networks:
+      - xain-fl-dev
```
Comment (little-dude) [inline docker-compose.yml]: shouldn't this be `xain-fl`?

File: docker-compose.yml [hunk 2]
```diff
@@ -19,6 +19,17 @@ services:
...
+    ports:
+      - "6379:6379"
```
Comment (Robert-Steiner) [inline docker-compose.yml]: Isn't `expose` better here instead of `ports` (like we did with minio)?

File: docker-compose.yml (hunk 1 above)
Comment (rsaffi) [reply docker-compose.yml]: Most definitely. That's called bad-copy-and-paste! :man_facepalming:

File: docker-compose.yml (hunk 2 above)
Comment (rsaffi) [reply docker-compose.yml]: Yes! For `docker-compose.yml` at least, for sure. For development file it makes sense for making it easier for DEVs to access Redis directly from the host, but for production environment, it should be kept restricted to local Docker network. Fixing it!

File: docker-compose-dev.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version: "3.7"
+version: "3"
```
Comment (Robert-Steiner) [inline docker-compose-dev.yml]: Why is it necessary to switch the version here but not in the release file?

Comment (rsaffi) [reply docker-compose-dev.yml]: That change was staged to be commited by mistake. I was testing something between v3 and v3.7 and commited that by mistake. Will revert it now.

**Pull request merged**